Skeleton/src/Volunteer.cpp

```diff
@@ -1 +1,263 @@
+{
+    return (activeOrderId != NO_ORDER);
+}
+
+
+/////////////////////////////////////////////////////////////////////////////////
+//////////////////////////// Collector Volunteer ////////////////////////////////
+/////////////////////////////////////////////////////////////////////////////////
+
+
+CollectorVolunteer::CollectorVolunteer(int id, const string &name, const int coolDown) : Volunteer(id, name) ,coolDown(coolDown), timeLeft(NO_ORDER){}
+CollectorVolunteer* CollectorVolunteer::clone() const 
+{
+     return new CollectorVolunteer(*this);
+}
+void CollectorVolunteer::step() // need to check if it's correct
+{
+    if(decreaseCoolDown())
+    {
+        completedOrderId = activeOrderId;
+        activeOrderId = NO_ORDER;
+    }
+}
+int CollectorVolunteer::getCoolDown() const
+{
+    return coolDown;
+}
+
+int CollectorVolunteer::getTimeLeft() const 
+{
+    return timeLeft;
+}
+
+bool CollectorVolunteer::decreaseCoolDown() //Decrease timeLeft by 1,return true if timeLeft=0,false otherwise
+{
+    if(timeLeft > 0){
+        timeLeft = timeLeft - 1;
+    }
+    return (timeLeft == 0);
+}
+
+bool CollectorVolunteer::hasOrdersLeft() const  // Signal whether the volunteer didn't reach orders limit,Always true for CollectorVolunteer and DriverVolunteer
+{
+    return true;
+} 
+
+bool CollectorVolunteer::canTakeOrder(const Order &order) const 
+{
+    return (order.getStatus() == OrderStatus::PENDING && !Volunteer::isBusy());
+}
+
+void CollectorVolunteer::acceptOrder(const Order &order) 
+{
+    if(canTakeOrder(order)){
+        activeOrderId = order.getId();
+        timeLeft = coolDown;
+    }
+}
+string CollectorVolunteer::toString() const 
+{
+    return "Collector Volunteer - ID: " + std::to_string(getId())
+     + ", Name: " + getName() 
+     + ", Active Order ID: " + std::to_string(getActiveOrderId()) 
+     + ", Completed Order ID: " + std::to_string(getCompletedOrderId());
+}
+
+
+//////////////////////////////////////////////////////////////////////////////////////////////
+//////////////////////////// LimitedCollector Volunteer //////////////////////////////////////
+//////////////////////////////////////////////////////////////////////////////////////////////
+
+
+LimitedCollectorVolunteer::LimitedCollectorVolunteer(int id, const string &name, int coolDown ,int maxOrders) : CollectorVolunteer(id, name, coolDown), maxOrders(maxOrders), ordersLeft(maxOrders) {}
+LimitedCollectorVolunteer* LimitedCollectorVolunteer::clone() const 
+{
+    return new LimitedCollectorVolunteer(*this);
+}
+
+bool LimitedCollectorVolunteer::hasOrdersLeft() const 
+{
+    return (ordersLeft > 0);
+}
+
+bool LimitedCollectorVolunteer::canTakeOrder(const Order &order) const 
+{
+    return (order.getStatus() == OrderStatus::PENDING && !CollectorVolunteer::isBusy() && hasOrdersLeft());
+}
+
+void LimitedCollectorVolunteer::acceptOrder(const Order &order) 
+{
+    if(canTakeOrder(order)){
+        ordersLeft--;
+        activeOrderId = order.getId();
```

Comment:
need to add timeleft = cooldown?

Reply:
not sure if to call the method 'acceptOrder' from the 'CollectorVolunteer' class
I mean to add the following code before the 'if' condition:
**CollectorVolunteer::acceptOrder(order);**

Reply:
i think you should only use this class method canTakeOrder and then implement the things inside.
you could use CollectorVolunteer::canTakeOrder(order) in the canTakeOrder method and add one exta condition.
but in this case i think its better to do it as is, its not complicated method